integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java

```diff
@@ -218,6 +218,42 @@ public static CoreV1Event getEvent(String opNamespace, String domainNamespace, S
     return null;
   }
 
+  /**
+   * Get matching event object with specific reason and type.
+   * @param domainNamespace namespace in which the event is logged
+   * @param domainUid UID of the domain
+   * @param reason event to check for Created, Changed, deleted, processing etc
+   * @param type type of event, Normal or Warning
+   * @param timestamp the timestamp after which to see events
+   * @return CoreV1Event matching event object
+   */
+  public static List<CoreV1Event> getEvents(String domainNamespace,
+                                            String domainUid,
+                                            String reason,
+                                            String type,
+                                            OffsetDateTime timestamp) {
+
+    List<CoreV1Event> events = new ArrayList<>();
+
+    try {
+      List<CoreV1Event> allEvents = Kubernetes.listNamespacedEvents(domainNamespace);
+      for (CoreV1Event event : allEvents) {
+        Map<String, String> labels = event.getMetadata().getLabels();
+        if (event.getReason().equals(reason)
+            && (isEqualOrAfter(timestamp, event))
+            && event.getType().equals(type)
+            && labels.containsKey("weblogic.createdByOperator")
+            && labels.get("weblogic.domainUID").equals(domainUid)) {
```

Comment:
The list contains all of the events in the namespace, we don't know if event event has labels. So we need to make sure that the "labels" variable is not null. Not sure if it is easier to add the listOpGeneratedNamespacedEvents method into release/3.3.

Reply:
fixed